### ISSUE TITLE
txrequest announcement warning

### ISSUE THREAD

**Issue by sidhujag** [bitcoin/bitcoin#20173]
Compiling master on bionic x86_64 I get this warning:

```
txrequest.cpp:73:21: warning: ‘{anonymous}::Announcement::m_state’ is too small to hold all values of ‘enum class {anonymous}::State’
     State m_state : 3;
```
                     
Do we do anything about it? You would think 3 bits (8 values) should be enough to store m_state which only has 5 values but maybe because the State enum is a uint8_t its complaining about possibly being too small.

**Comment by fanquake** [comment]
This is being fixed in #20162.

**Comment by sidhujag** [comment]
ACK
